### PR TITLE
[32] add map_row and map_blocks for pandas dataframe

### PR DESCRIPTION
**Pull request by yupbank**
this pr add `map_row` and `map_blocks` support for pandas dataframe 
```
        data = [dict(x=float(x)) for x in range(10)]
        df = pd.DataFrame(data)
        with tf.Graph().as_default() as g:
            # The placeholder that corresponds to column 'x'
            x = tf.placeholder(tf.double, shape=[None], name="x")
            # The output that adds 3 to x
            z = tf.add(x, 3, name='z')
            # The resulting dataframe
            df1 = tfs.map_blocks(z, df)
            df2 = tfs.map_row(z, df)
            y =  tf.placeholder(tf.double, shape=[None], name="y")
            u = tf.add(y, 3, name='u')
            df3 = tfs.map_row(u, df, feed_dict={'y': 'x'})
            df4 = tfs.map_blocks(u, df, feed_dict={'y': 'x'})
```

### CODE REVIEW

File: src/main/python/tensorframes/core.py [hunk 1]
```diff
@@ -134,6 +136,33 @@ def _map(fetches, dframe, feed_dict, block, trim):
...
+def _get_input(graph):
+    ph_names = []
+    for n in graph.as_graph_def().node:
+        # Just the input nodes:
+        if n.op == 'Placeholder':
+            ph_names.append(n.name)
+    return ph_names
+
+def _unique_graph(fetches):
+    graph = set()
```
Comment (thunterdb) [inline src/main/python/tensorframes/core.py]: `graph = set([t.graph for t in fetches])`?

File: src/main/python/tensorframes/core.py [hunk 2]
```diff
@@ -208,6 +237,8 @@ def map_rows(fetches, dframe, feed_dict=None):
     :param dframe: a Spark DataFrame
     :return: a Spark DataFrame
     """
+    if isinstance(dframe, pd.DataFrame):
```
Comment (thunterdb) [inline src/main/python/tensorframes/core.py]: Can you update the documentation above to reflect that the input can be either pandas or spark dataframes?

**Comment by yupbank** [comment]
>I just have two small comments, otherwise it looks great. Thanks for taking a look at this issue!
Show outdated

hey how is it going?

**Comment by thunterdb** [comment]
@yupbank thanks for addressing the comments! I am merging.

**Pull request merged**